utils/python/CIME/case_run.py

```diff
@@ -0,0 +1,279 @@
+from CIME.task_maker                import TaskMaker
+
+import gzip, shutil, time, sys, os
+
+logger = logging.getLogger(__name__)
+
+###############################################################################
+def preRunCheck(case):
+###############################################################################
+
+    # Pre run initialization code..
+    caseroot = case.get_value("CASEROOT")
+    cimeroot = case.get_value("CIMEROOT")
+    din_loc_root = case.get_value("DIN_LOC_ROOT")
+    compiler = case.get_value("COMPILER")
+    debug = case.get_value("DEBUG")
+    mach = case.get_value("MACH")
+    batchsubmit = case.get_value("BATCHSUBMIT")
+    mpilib = case.get_value("MPILIB")
+    rundir = case.get_value("RUNDIR")
+    build_complete = case.get_value("BUILD_COMPLETE")
+
+    # check for locked files.
+    check_lockedfiles()
+    logger.debug("check_lockedfiles OK")
+
+    # check that build is done
+    expect (build_complete,
+            "BUILD_COMPLETE is not true\nPlease rebuild the model interactively")
+    logger.debug("build complete is %s " %build_complete)
+
+    # load the module environment...
+    env_module = EnvModule(mach, compiler, cimeroot, caseroot, mpilib, debug)
+    env_module.load_env_for_case()
+
+    # set environment variables
+    # This is a requirement for yellowstone only
+    if mpilib == "mpi-serial" and "MP_MPILIB" in os.environ:
+        del os.environ["MP_MPILIB"]
+    else:
+        os.environ["MPILIB"] = mpilib
+
+    if batchsubmit is None or len(batchsubmit) == 0:
+        os.environ["LBQUERY"] = "FALSE"
+        os.environ["BATCHQUERY"] = "undefined"
+    elif batchsubmit == 'UNSET':
+        os.environ["LBQUERY"] = "FALSE"
+        os.environ["BATCHQUERY"] = "undefined"
+    else:
+        os.environ["LBQUERY"] = "TRUE"
+
+    # create the timing directories, optionally cleaning them if needed.
+    if not os.path.isdir(rundir):
+        os.mkdir(rundir)
+
+    if os.path.isdir(os.path.join(rundir,"timing")):
+        shutil.rmtree(os.path.join(rundir,"timing"))
+
+    os.makedirs(os.path.join(rundir,"timing","checkpoints"))
+
+    # run preview namelists
+    preview_namelists()
+
+    # document process
+    append_status("Run started ",caseroot=caseroot,
+                 sfile="CaseStatus")
+
+    logger.info( "-------------------------------------------------------------------------")
+    logger.info( " - To prestage required restarts, untar a restart.tar file into %s" %(rundir))
+    logger.info( " - Case input data directory (DIN_LOC_ROOT) is %s " %(din_loc_root))
+    logger.info( " - Checking for required input datasets in DIN_LOC_ROOT")
+    logger.info( "-------------------------------------------------------------------------")
+
+###############################################################################
+def runModel(case):
+###############################################################################
+
+    # Set OMP_NUM_THREADS
+    tm = TaskMaker(case=case)
```

Comment:
Changes here to avoid having to do token replacement. Used to have {{ thread_count }} and later {{ mpirun }}.